modules/apps/27-interchain-accounts/host/keeper/keeper.go

```diff
@@ -106,6 +109,22 @@ func (k Keeper) GetAppVersion(ctx sdk.Context, portID, channelID string) (string
 	return k.ics4Wrapper.GetAppVersion(ctx, portID, channelID)
 }
 
+// GetAppMetadata retrieves the interchain accounts metadata from the store associated with the provided portID and channelID
+func (k Keeper) GetAppMetadata(ctx sdk.Context, portID, channelID string) (icatypes.Metadata, error) {
```

Comment:
If there's no usage for external apps to use this function, I think it would be possible to make it unexported and make a wrapper around it to be able to use it in the tests.

Reply:
This is used in [`modules/apps/27-interchain-accounts/host/keeper/relay.go:27`](https://github.com/cosmos/ibc-go/pull/3796/files#diff-38cebbb99f68c1e182e3752561d661da36713b1a10932f6fca035998d94ae185R27)

Reply:
Yeah, I saw where it's used, and I am not proposing to eliminate the function. I am just suggesting that it might be possible to make the function unexported, so that we don't increate our public API (assuming that there is no reason for external applications to use this function).

Reply:
Sorry for my misunderstanding. I think `GetAppMetadata` is as useful if not more than `GetAppVersion`. Also pretty much all other functions in the keeper are public. Also how do you put a wrapper around it, do you have an example?

Reply:
You can find one [here](https://github.com/cosmos/ibc-go/pull/3895/files). Fine to leave it public if there's going to be a use for it, but if it's not clear at the moment maybe is better to make it private and only expose it when requested?

Reply:
I agree with @crodriguezvega. Historically a lot of functions have been made public that don't need to be, but we have made a concerted effort to decrease the API so it is very clear which functions are intended for external usage